lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -8,25 +8,14 @@ include_once 'dosomething_kudos.kudos.inc';
 
 
 /**
- * @param $parameters
- * @return array  IDs of newly created records.
+ * Create a Kudos record.
+ *
+ * @param $values array Associated array of database column name and value pairs.
+ * @return DatabaseStatementInterface|int
  * @throws Exception
  */
-function dosomething_kudos_create($parameters) {
-  $records = array();
-
-  $values = array(
-    'fid' => $parameters['fid'],
-    'uid' => $parameters['uid'],
-  );
-
-  foreach($parameters['tids'] as $id) {
-    $values['tid'] = $id;
-
-    $records[] = db_insert('dosomething_kudos')->fields($values)->execute();
-  }
-
-  return $records;
+function dosomething_kudos_create($values) {
+  return db_insert('dosomething_kudos')->fields($values)->execute();
```

Comment:
Discussed this w @weerd. We're going to go with the approach of creating a Kudos Entity to handle these CRUD functions, to keep consistent with how entities like Signups, Reportbacks, Reportback Files, Shipments, Rewards, etc work.

Ideally the `Kudos` class created can be used for the definition in `hook_entity_info`. Consistent naming convention would call the class `KudosEntity` ... but, we can worry about that once you get there...